### ISSUE TITLE
usability issue with ipfs block put

### ISSUE THREAD

**Issue by whyrusleeping** [jbenet/go-ipfs#769]
Working on more examples, ipfs block put is relatively hard to use, Setting the data field requires using base64, which really prevents me from just hacking together a dag object in vim like:
```
{
    "Data": "Hello world!"
}
```
Im not sure what the right solution to this is... but its kindof obnoxious to work with.

**Comment by jbenet** [comment]
I agree, it's painful. What's a viable alternative? that's how json handles arbitrary binary input.

**Comment by whyrusleeping** [comment]
Well, as obnoxious as it is, if we print the json struct using a string instead of a []byte, it will use a unicode escape character type encoding, i.e.:

```
{"B":"AQIDBAUGBw=="}
```
becomes:
```
{"B":"\u0001\u0002\u0003\u0004\u0005\u0006\u0007"}
```

**Comment by jbenet** [comment]
I'm not sure that's better. both suck. :( -- for users of libraries, this is not a big problem. it's mostly just cli usage.

**Comment by whyrusleeping** [comment]
Well, its better at least because it allows me to write real text in the string without worrying about base64. i could write:
```
{"B":"Hello World Heres a weird character:\u0007"}
```

**Comment by jbenet** [comment]
how does that work with quotes?

**Comment by whyrusleeping** [comment]
it just backslash escapes them

**Comment by kevinwallace** [comment]
What about byte strings that are not valid UTF-8?  Or are you using the string as a container for code points only in the range `\u0000`-`\u00FF`, and layering any unicode encoding on top of that?

**Comment by whyrusleeping** [comment]
@kevinwallace, what byte strings would those be? we would essentially be casting a `[]byte` to a `string` at the last moment, giving us 256 possible characters to be printed. I beleive that limits us to the range `\u0000` - `\u00FF`

**Comment by kevinwallace** [comment]
@whyrusleeping Sorry, my bad.  I thought Go gave special treatment to `string`<->`[]byte` casts, but actually a string contains an arbitrary byte array.  It's the `string`<->`[]rune` cast that involves UTF-8 conversion.  See <http://blog.golang.org/strings>.  I don't think this is unsafe anymore.

**Comment by whyrusleeping** [comment]
Yeah, not gonna try and deal with runes, lol. the only reason for involving strings is changing how the json library outputs the info.

**Comment by jbenet** [comment]
closed in #810
